specs/v2-sketch.yaml

```diff
@@ -0,0 +1,365 @@
+          deprecated: true
+        enrolled_betas:
+          type: array
+          items:
+            type: string
+        analytics_id:
+          type: string
+          format: uuid
+          description: can we call this "id", please?
+        student:
+          type: boolean
+          default: false
+        privacy-optout:
+          type: boolean
+          description: Opt out of analytics trackers
+#concurrency
+#   Appears unused
+#
+#encrypted-heroku-api-key
+#   Nope, not including credentials, encrypted or not.
+#
+#plans
+#   Not sure what these should be
+        containers:
+          type: number
+          format: integer
+        num_projects_followed:
+          type: number
+          format: integer
+        days_left_in_trial:
+          type: number
+          format: integer
+        all_emails:
+          type: array
+          example: ["username@example.com"]
+          items:
+            type: string
+        selected_email:
+          type: string
+          example: "username@example.com"
+        avatar_url:
+          type: string
+          example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+# github_id
+# bitbucket
+# bitbucket-authorized
+# pusher-id
+#   These all feel far too specific, should be rolled into "identities", right?
+#
+# github-oauth-scopes
+#   don't wanna include GitHub oauth scopes
+#
+# heroku-api-key
+#   Nope, not including credentials, obfuscated or not
+        identities:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              avatar_url:
+                type: string
+                example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+              external_id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              name:
+                type: string
+                example: "User Name"
+              user?:
+                type: boolean
+              domain:
+                type: string
+                example: "github.com"
+              type:
+                type: string
+                example: "github"
+              authorized?:
+                type: boolean
+              provider_id:
+                type: string
+                format: uuid
+              login:
+                type: string
+                example: "username"
+    Actor:
+      type: object
+      properties:
+        login:
+          type: string
+        name:
+          type: string
+        id:
+          type: string
+          format: uuid
+    Project:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+        # add a link to the project, Pat suggested we consider link headers
```

Comment:
Some sort of consistent approach to linking to related entities would be good. Link headers are fine. HATEOAS-style `_links` maps are fine. I suppose even inlining it like this is fine. I really just wanted to drive some conversation about how we do this and then do it consciously.

Reply:
Just including the ID is possible, but less pleasant for an API consumer. I'll write up a doc to drive some discussion.